File: src/Application Specific/Classes/Entity Collections/Checklist/SubTask/SubTask.cs
using System.Collections;

namespace FSM.Entity
{
    namespace SubTasks
    {
        public class SubTask
        {
            private DataTypeValidator _dataTypeValidator;

            public SubTask()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _SubTaskID = 0;

            public int SubTaskID
            {
                get
                {
                    return _SubTaskID;
                }
            }

            public object SetSubTaskID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SubTaskID", value);
                }
            }

            private int _TaskID = 0;

            public int TaskID
            {
                get
                {
                    return _TaskID;
                }
            }

            public object SetTaskID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TaskID", value);
                }
            }

            private string _Description = string.Empty;

            public string Description
            {
                get
                {
                    return _Description;
                }
            }

            public object SetDescription
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Description", value);
                }
            }

            private int _OrderNumber = 0;

            public int OrderNumber
            {
                get
                {
                    return _OrderNumber;
                }
            }

            public object SetOrderNumber
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderNumber", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Checklist/SubTask/SubTask.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace SubTasks

Public Class SubTask

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "SubTask Properties"

      
Private _SubTaskID As Integer = 0
Public Readonly Property SubTaskID() As Integer
	Get 
		Return _SubTaskID
	End Get	
End Property	
Public Writeonly Property SetSubTaskID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_SubTaskID", Value)
	End Set
End Property


Private _TaskID As Integer = 0
Public Readonly Property TaskID() As Integer
	Get 
		Return _TaskID
	End Get	
End Property	
Public Writeonly Property SetTaskID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_TaskID", Value)
	End Set
End Property


Private _Description As String = String.Empty
Public Readonly Property Description() As String
	Get 
		Return _Description
	End Get	
End Property	
Public Writeonly Property SetDescription() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Description", Value)
	End Set
End Property


Private _OrderNumber As Integer = 0
Public Readonly Property OrderNumber() As Integer
	Get 
		Return _OrderNumber
	End Get	
End Property	
Public Writeonly Property SetOrderNumber() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_OrderNumber", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

